pkg/network/common/egressip.go

```diff
@@ -290,6 +300,153 @@ func (eit *EgressIPTracker) UpdateHostSubnetEgress(hs *networkv1.HostSubnet) {
 	eit.syncEgressIPs()
 }
 
+func nodeIsReady(node *v1.Node) bool {
+	for _, cond := range node.Status.Conditions {
+		if cond.Type == v1.NodeReady {
+			if cond.Status == v1.ConditionFalse || cond.Status == v1.ConditionUnknown {
+				return false
+			}
+		}
+	}
+	return true
+}
+
+// SetMonitorNodes updates EgressIPTracker with the node data provided and
+// starts a go-routine (if one is not already running) which polls monitorNodes
+// to check if they are ready and reachable
+func (eit *EgressIPTracker) SetMonitorNodes(monitorNodes map[string]*NodeEgress) {
+	eit.Lock()
+	defer eit.Unlock()
+	eit.monitorNodes = monitorNodes
+	if len(monitorNodes) > 0 {
+		if eit.stop == nil {
+			eit.stop = make(chan struct{})
+			go utilwait.PollUntil(defaultPollInterval, eit.poll, eit.stop)
+		}
+	} else {
+		if eit.stop != nil {
+			close(eit.stop)
+			eit.stop = nil
+		}
+	}
+}
+
+// GetMonitorNodes gets the node data from EgressIPTracker
+func (eit *EgressIPTracker) GetMonitorNodes() map[string]*NodeEgress {
+	eit.Lock()
+	defer eit.Unlock()
+	return eit.monitorNodes
+}
+
+// AddEgressIP assumes the EgressIPTracker lock is already held. The access to
+// this function call is currently percolated down from `syncEgressIPs`, which
+// always has the lock held. It starts a go-routine (if one is not already
```

Comment:
Ugh. This is pretty sketchy; an exported API shouldn't be making assumptions about its caller like that...
(I agree that the current state of the code is _correct_; it's just not very good API design on our part.)
Maybe leave a FIXME comment here pointing out that this could be made better?

Reply:
I will add a FIXME